src/front/zokrates/mod.rs

```diff
@@ -520,14 +523,24 @@ impl<'ast> ZGen<'ast> {
                     };
                     self.circ.cir_ctx().cs.borrow_mut().outputs.push(cmp);
                 }
+                Mode::Fhe => {
+                    let ret_term = r.unwrap_term();
+                    let ret_terms = ret_term.terms();
+                    self.circ
+                        .cir_ctx()
+                        .cs
+                        .borrow_mut()
+                        .outputs
+                        .extend(ret_terms);
+                }
```

Comment:
How about you merge this case with that for `Mode::Mpc` using an [or pattern](https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#multiple-patterns)?